### PR TITLE
Knee walls

### PR DESCRIPTION
**Pull request by nmerket**
## Pull Request Description

Adds knee walls support.

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [x] Code changes (must work)
- [x] Test exercising your feature or bug fix. Check the coverage report in the build artifacts.
- [x] All other unit tests passing
- [x] Update translation docs

### CODE REVIEW

**Comment by nmerket** [comment]
Question: House 5 has a vented attic and roof insulation. Should we allow that? I'm parsing out the attic ceiling area.

File: hescorehpxml/schemas/hescore_json.schema.json [hunk 1]
```diff
@@ -272,7 +275,7 @@
                                     },
                                     "roof_type": {
                                         "type": "string",
-                                        "enum": ["vented_attic", "cond_attic", "cath_ceiling"],
+                                        "enum": ["vented_attic", "cath_ceiling"],
```
Comment (bpark1327) [inline hescorehpxml/schemas/hescore_json.schema.json:278]: Is this change intentional?

File: hescorehpxml/schemas/hescore_json.schema.json [hunk 2]
```diff
@@ -389,72 +411,65 @@
...
+                                    "roof_type",
+                                    "roof_assembly_code",
+                                    "roof_color"
+                                ],
                                 "allOf": [
                                     {
                                         "if": {
                                             "properties": {
-                                                "roof_area": {"exclusiveMinimum": 0}
+                                                "roof_type": {"enum": ["vented_attic"]}
```
Comment (bpark1327) [inline hescorehpxml/schemas/hescore_json.schema.json:424]: The use of `const` vs `enum` with a single value is inconsistent. Probably we should use just `const` unless there is a good reason.

**Review by bpark1327 — COMMENTED** [review]
Looks great to me! I have a couple of minor comments/questions.

File: docs/source/translation/zone_roof.rst [hunk 3]
```diff
@@ -82,6 +66,68 @@ attic. This is discussed in more detail in :ref:`roof-rvalues`.
...
+Vented Attic
+------------
+
+The thermal boundary for a vented attic is at the floor of the attic. In **HPXML
+v2** that area is attached to the
+``Building/BuildingDetails/Enclosure/AtticAndRoof/Attics/Attic`` element:
+
+.. literalinclude:: ../../../examples/hescore_min.xml
+   :lines: 65-82
+   :emphasize-lines: 17
```
Comment (shorowit) [inline docs/source/translation/zone_roof.rst:91]: @nmerket This is nice, but won't this get out of date every time the example files are updated? If so, I assume it would be really easy to not notice that the wrong lines are suddenly being pointed to.

File: examples/house9.json [hunk 4]
```diff
@@ -23,12 +23,16 @@
       "zone_roof": [
         {
           "roof_name": "roof1",
-          "roof_area": 2176.0,
           "roof_assembly_code": "rfwf00wo",
           "roof_color": "cool_color",
-          "roof_absorptance": 0.85,
+          "roof_absorptance": 0.8500000000000001,
```
[the remark this answers is not in the document]
Comment (nmerket) [reply examples/house9.json:28]: Yeah, I suppose this is just what came out of the translator. I can look into rounding.

File: hescorehpxml/schemas/hescore_json.schema.json (hunk 1 above)
Comment (nmerket) [reply hescorehpxml/schemas/hescore_json.schema.json:278]: Yes, we decided to not support cond_attic anymore.

File: hescorehpxml/schemas/hescore_json.schema.json (hunk 2 above)
Comment (nmerket) [reply hescorehpxml/schemas/hescore_json.schema.json:424]: So, would it look like this?

```suggestion
                                                "roof_type": {"const": "vented_attic"}
```

File: docs/source/translation/zone_roof.rst (hunk 3 above)
Comment (nmerket) [reply docs/source/translation/zone_roof.rst:91]: Yes, it does get outdated. I do this in a couple of other places as well. These blocks get outdated either way and need to be updated.
